edx-solutions/xblock-drag-and-drop-v2
# Assessment mode: display item feedback

**Note:** ~~This PR is based on #87, only relevant commit is the last one.~~

**Description:** This PR allows DnDv2 to show item feedback messages in assessment mode.

**JIRA:** [SOL-1980](https://openedx.atlassian.net/browse/SOL-1980)

**Discussions:** [SOL-1980](https://openedx.atlassian.net/browse/SOL-1980)

**Dependencies:** ~~#87~~

**Sandboxes:**

* LMS: http://dndv2-sandbox8.opencraft.hosting/
* Studio: http://studio-dndv2-sandbox8.opencraft.hosting

**Testing instructions:**
1. Configure DnDv2 to have unique "incorrect" item feedback messages. Set to assessment mode.
2. In LMS, put some items into *wrong* zones.
3. Check answer (click "Submit")
4. Feedback popup should appear showing incorrect feedback messages for every misplaced item.
5. ~~If only one item is misplaced, only feedback message is shown; if multiple items are misplaced,~~ popup contains a note about ~~multiple~~ item(s) being misplaced, and feedback messages are shown in form of bullet list.

**Screenshots:**







![image](https://cloud.githubusercontent.com/assets/3330048/17589879/bf581ad8-5fdf-11e6-85b1-1cc734e444b3.png)

**Reviewers:**
- [x] internal: @itsjeyd 
- [x] internal: @haikuginger
- [x] TNL: @cahrens 
- [x] TNL: @staubina
- [ ] ~~a11y: @cptvitamin~~ a11y review will be performed later, for all changes in v2.1: [SOL-1978](https://openedx.atlassian.net/browse/SOL-1978)

**Checklist:**
- [x] Internal review
- [x] Upstream review
